test/dynamodb/stream-write.test.js

```diff
@@ -0,0 +1,168 @@
+const set = require('lodash/set');
+const sinon = require('sinon');
+const test = require('ava');
+const util = require('util');
+const uuid = require('uuid/v4');
+
+const { dynamodb } = require('../..');
+const { requestToDeleteItem, requestToUpdateItem, mockItemQueryStream } = require('./helpers/streams');
+const { tap, toArray } = require('rxjs/operators');
+
+const assertItemDeleted = (stub, table, item) => sinon.assert.calledWithExactly(
+  stub,
+  sinon.match(requestToDeleteItem(table, item.id)),
+  sinon.match.func
+);
+
+const assertItemUpdated = (stub, table, item) => sinon.assert.calledWithExactly(
+  stub,
+  sinon.match(requestToUpdateItem(table, item)),
+  sinon.match.func
+);
+
+const mockMethod = (documentClient, name) => {
+  const stub = sinon.stub().yields(null, {});
+  const promisified = util.promisify(stub);
+
+  const method = (...args) => ({
+    promise: () => promisified(...args)
+  });
+
+  set(documentClient, name, method);
+  return stub;
+};
+
+const testWritingItems = async (test, {operator, validator}) => {
+  const { batchWrite, dynamoClient } = test.context;
+  const evenTable = uuid();
+  const oddTable = uuid();
+
+  const items = new Array(42).fill(null)
+    .map((item, index) => ({
+      item: { id: uuid() },
+      table: index % 2 ? oddTable : evenTable
+    }));
+
+  const {write, writeBufferSpace} = dynamodb.write(dynamoClient);
+
+  const expected = [];
+  const performed = await mockItemQueryStream(items, writeBufferSpace)
+    .pipe(operator)
```

Comment:
Any reason you used so many calls to `.pipe` in a row instead of the multi-arg pipe that other code uses?

Reply:
Nope. It was just already this way and was working. Should be easy enough to cleanup.